src/components/EditPhoto.jsx

```diff
@@ -28,10 +28,14 @@ export const EditPhoto = ({ handleUpdate, showInput, shouldShowInput }) => {
       <Button
         variant="contained"
         color="success"
-        style={{margin:"10px"}}
+        style={{ margin: "10px" }}
         onClick={() => {
-          handleUpdate(imageUrl);
-          setImageUrl("");
+          if (new URL(url)) {
+            handleUpdate(imageUrl);
+            setImageUrl("");
+          } else {
+            throw new Error("Invalid URL");
```

Comment:
you should create a state to show what error the user gets and show it below the input. Also is good to encapsulate it on a external function